index.js

```diff
@@ -64,30 +64,37 @@ exports.indexLinks = function (message, opts, each) {
     opts = { rel: opts }
   if (!opts)
     opts = {}
-  var msg  = opts.msg
+  var msg = opts.msg
   var feed = opts.feed
   var blob = opts.blob
-  var any  = !(msg || feed || blob)
+  var any = !(msg || feed || blob)
 
   traverse(toMsgContent(message), function (obj, rel) {
     if (opts.rel && rel !== opts.rel) return
 
-    var r = (typeof obj == 'string') ? obj : obj.link
+    var r;
+    if (typeof obj == 'string') {
+      r = obj
+    } else if (typeof obj.link == 'string') {
```

Comment:
I just tested this and the server still crashes over this line right here.

The content of `message` is:

```
{ type: 'post',
  text: '[SoapDog (SurfaceGo)](@qv10rF4IsmxRZb7g5ekJ33EakYBpdrmV/vtP1ij5BS4=.ed25519) do I have a bug in mentions? #patchfox',
  root: '%oxXKoCVtqM111XqKyTyTOCXt6s/51/xI01O8F7DBbRQ=.sha256',
  branch: '%K+hbT4mTrcGgg17kwt86aoimRBtFHry1bcJFDdY7SlU=.sha256',
  mentions:
   [ { link: '@qv10rF4IsmxRZb7g5ekJ33EakYBpdrmV/vtP1ij5BS4=.ed25519',
       name: 'SoapDog (SurfaceGo)' },
     null ],
  channel: 'patchfox' }
```
Apparently one of the mentions in the message is a nul (and somehow my database was fine pulling that down somehow)

the content of `obj` is: `null`
the content of `rel` is: `mentions`

The error produced is:

```
/home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-msgs/index.js:81
    } else if (typeof obj.link == 'string') {
                          ^

TypeError: Cannot read property 'link' of null
    at /home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-msgs/index.js:81:27
    at /home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-msgs/index.js:50:9
    at Array.forEach (<anonymous>)
    at traverse (/home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-msgs/index.js:49:11)
    at Object.exports.indexLinks (/home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-msgs/index.js:72:3)
    at indexMsg (/home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-db/indexes/links.js:25:10)
    at /home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-db/indexes/links.js:34:12
    at reduce (/home/node/.npm-global/lib/node_modules/ssb-server/node_modules/ssb-db/node_modules/flumeview-level/index.js:90:25)
    at /home/node/.npm-global/lib/node_modules/ssb-server/node_modules/pull-write/index.js:35:23
    at /home/node/.npm-global/lib/node_modules/ssb-server/node_modules/pull-write/index.js:41:13
```

I'm unsure what to do with this

Reply:
an easy fix would be to:

```
} else if (obj && obj.link == 'string') {
```

But I doubt if we shouldn't fix things higher up the stack instead

Reply:
amending the PR...

Reply:
I've fixed with the code you've sent.